modules/consentManagement.js

```diff
@@ -100,11 +100,7 @@ function lookupIabConsent(cmpSuccess, cmpError, hookConfig) {
   function v2CmpResponseCallback(tcfData, success) {
     utils.logInfo('Received a response from CMP', tcfData);
     if (success) {
-      if (tcfData.eventStatus === 'tcloaded' || tcfData.eventStatus === 'useractioncomplete') {
-        cmpSuccess(tcfData, hookConfig);
-      } else if (tcfData.eventStatus === 'cmpuishown' && tcfData.tcString && tcfData.purposeOneTreatment === true) {
-        cmpSuccess(tcfData, hookConfig);
-      }
+      cmpSuccess(tcfData, hookConfig);
```

Comment:
when the `eventStatus` === "cmpuishown" we may not have a valid `tcString` yet, like we do with "tcloaded" and "useractioncomplete".  I don't recall or understand why `tcfData.purposeOneTreatment` is significant here, but I don't think we want to call `cmpSuccess()` if we're still waiting for a valid `tcString`.

Reply:
The previous implementation calls `cmpSuccess(tcfData, hookConfig)` in all cases.



> I don't think we want to call `cmpSuccess()` if we're still waiting for a valid `tcString`.

A CMP returns a "Valid" TC string in all `eventStatus`es.  Once `"cmpuishown"` is triggered legitimate interest has been established but consent hasn't been gathered.  Some "Vendors" only rely on that legal basis.  Vendors should be able to determine which state they need the TCString to be in.

Reply:
I incorrectly said the `tcString` may not be "valid".  It's valid, but user consent may not have been gathered yet.   I believe `cmpSuccess()` is intended to be called when the user's consent preferences have been obtained.  Most bidders will want the user's consent, if possible, before sending a bid request.  Perhaps a change could be made to give the publisher the option to proceed with the auction without giving the user a chance to respond.  Maybe setting the timeout to zero already accomplishes this?

Reply:
There is more nuance to TCFv2 in that a legitimate interest basis for processing personal data as defined by [GDPR Article 6(1)(f)](https://gdpr-info.eu/art-6-gdpr/) is now represented in the TC String.  There are Vendors who operate solely on Legitimate Interest and that's why these events were created.  It may be true that most bidders will want the user's consent but there are some who were advocating for this feature through the design of TCFv2.

Reply:
Prebid doesn't work that way.  There's no mechanism for some bidders to wait for additional consent from the user (or wait for anything else for that matter) while others do not.  All bidders are called at the same time and have the same amount of time to respond before being timed out.  If a publisher doesn't want bidders to wait for the user to respond, they could set the `cmpTimeout` to 0 (or another small number).  Though I guess in that case we wouldn't be using the default `tcData` that was returned (Though we may want to update the code to save it as a backup if we don't get a response from the other events before the timeout).  There may also be some GDPR Enforcement repercussions to be considered (when that module is included), if we only get a default tcData object.